plugins/net_plugin/net_plugin.cpp

```diff
@@ -3991,6 +4071,54 @@ namespace eosio {
       return fc::json::from_string(s).as<T>();
    }
 
+   std::tuple<std::string, size_t> net_plugin_impl::parse_listen_address( const std::string& address ) const {
+      auto listen_addr = address;
+      auto limit = string("0");
+      auto last_colon_location = address.rfind(':');
+      if( auto right_bracket_location = address.find(']'); right_bracket_location != address.npos ) {
+         if( std::count(address.begin()+right_bracket_location, address.end(), ':') > 1 ) {
+            listen_addr = std::string(address, 0, last_colon_location);
+            limit = std::string(address, last_colon_location+1);
+         }
+      } else {
+         if( auto colon_count = std::count(address.begin(), address.end(), ':'); colon_count > 1 ) {
+            EOS_ASSERT( colon_count <= 2, plugin_config_exception, "Invalid address specification ${addr}; IPv6 addresses must be enclosed in square brackets.", ("addr", address));
+            listen_addr = std::string(address, 0, last_colon_location);
+            limit = std::string(address, last_colon_location+1);
+         }
+      }
+      auto block_sync_rate_limit = parse_connection_rate_limit(limit);
+
+      return {listen_addr, block_sync_rate_limit};
+   }
+
+   size_t net_plugin_impl::parse_connection_rate_limit( const std::string& limit_str) const {
+      std::istringstream in(limit_str);
+      fc_dlog( logger, "parsing connection endpoint limit ${limit} with locale ${l}", ("limit", limit_str)("l", std::locale("").name()));
```

Comment:
Wouldn't parsing the config in a locale dependent way mean config files aren't transportable across different systems (when they have a different locale)? I can imagine some confusion from users when copying a sample config file and it errors out on their system.

Reply:
Removed locale-awareness.